src/main/java/org/opensearch/security/transport/SecurityInterceptor.java

```diff
@@ -270,33 +278,28 @@ public T read(StreamInput in) throws IOException {
 
         @Override
         public void handleResponse(T response) {
-            final List<String> flsResponseHeader = getThreadContext().getResponseHeaders().get(ConfigConstants.OPENDISTRO_SECURITY_FLS_FIELDS_HEADER);
-            final List<String> dlsResponseHeader = getThreadContext().getResponseHeaders().get(ConfigConstants.OPENDISTRO_SECURITY_DLS_QUERY_HEADER);
-            final List<String> maskedFieldsResponseHeader = getThreadContext().getResponseHeaders().get(ConfigConstants.OPENDISTRO_SECURITY_MASKED_FIELD_HEADER);
-
+        	
+            ThreadContext threadContext = getThreadContext();
+            Map<String, List<String>> responseHeaders = threadContext.getResponseHeaders();
+
+            List<String> flsResponseHeader = responseHeaders.get(ConfigConstants.OPENDISTRO_SECURITY_FLS_FIELDS_HEADER);
+            List<String> dlsResponseHeader = responseHeaders.get(ConfigConstants.OPENDISTRO_SECURITY_DLS_QUERY_HEADER);
+            List<String> maskedFieldsResponseHeader = responseHeaders.get(ConfigConstants.OPENDISTRO_SECURITY_MASKED_FIELD_HEADER);
+            
             contextToRestore.restore();
 
-            final boolean isDebugEnabled = log.isDebugEnabled();
-            if (response instanceof ClusterSearchShardsResponse && flsResponseHeader != null && !flsResponseHeader.isEmpty()) {
-                if (isDebugEnabled) {
-                    log.debug("add flsResponseHeader as transient");
```

Comment:
Why removing the debug logging?

Reply:
Good catch, this was done by mistake. Added the debug statements again